### PR TITLE
[RFC] api: nvim_buf_delete

### PR DESCRIPTION
**Pull request by tjdevries**

### CODE REVIEW

File: src/nvim/api/buffer.c [hunk 1]
```diff
@@ -953,6 +953,86 @@ Boolean nvim_buf_is_loaded(Buffer buffer)
   return buf && buf->b_ml.ml_mfp != NULL;
 }
 
+
+/// Unloads the buffer. See |:bunload|
+///
+/// @param buffer Buffer handle, or 0 for current buffer
+/// @param force  See <bang> description for |:bunload|
+void nvim_buf_unload(Buffer buffer, Boolean force, Error *err)
```
Comment (bfredl) [inline src/nvim/api/buffer.c:961]: do we really want three API functions that do almost the same thing? Can we not just a flag dict already

File: src/nvim/api/buffer.c [hunk 2]
```diff
@@ -953,6 +953,61 @@ Boolean nvim_buf_is_loaded(Buffer buffer)
   return buf && buf->b_ml.ml_mfp != NULL;
 }
 
+#define UNPACK_BOOL_RV(result, v, message, rv) \
```
Comment (bfredl) [inline src/nvim/api/buffer.c:956]: `api_is_truthy` ?

**Comment by bfredl** [comment]
rebase?

**Comment by tjdevries** [comment]
rebase what? You can just squash my commits here when we're good.

**Comment by bfredl** [comment]
the PR. i see unrelated doc changes

**Comment by tjdevries** [comment]
Ah, gotcha -- all set :)

File: src/nvim/api/buffer.c [hunk 3]
```diff
@@ -953,6 +953,53 @@ Boolean nvim_buf_is_loaded(Buffer buffer)
...
+
+  if (ERROR_SET(err)) {
+    return;
+  }
+
+  bool force = false;
+  bool unload = false;
+  for (size_t i = 0; i < opts.size; i++) {
+    String k = opts.items[i].key;
+    Object *v = &opts.items[i].value;
```
Comment (bfredl) [inline src/nvim/api/buffer.c:975]: unnecessary pointer indirection.

File: src/nvim/api/private/helpers.c [hunk 4]
```diff
@@ -1619,14 +1619,20 @@ VirtText parse_virt_text(Array chunks, Error *err)
   return virt_text;
 }
 
-bool api_is_truthy(Object obj, const char *what, bool nil_truthy, Error *err)
+/// Force obj to bool.
+/// If it fails, returns false and sets err
+/// @param obj      The object to coerce to a boolean
+/// @param what     The name of the object, used for error message
+/// @param if_nil   What to return if the type is nil.
```
Comment (bfredl) [inline src/nvim/api/private/helpers.c:1626]: `if_nil` sounds like "was the value nil". I like the old one better. or maybe `nil_value` (as boolean is a value in this context).

**Pull request merged**